staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -220,6 +227,11 @@ func LogAnnotation(ae *auditinternal.Event, key, value string) {
 	if ae == nil || ae.Level.Less(auditinternal.LevelMetadata) {
 		return
 	}
+	// In mutating webhooks we persist webhook patch in audit annotation with patchAuditAnnotationPrefix.
+	// Since the patch mutates request body, we skip logging if audit level is less than Request.
+	if strings.HasPrefix(key, patchAuditAnnotationPrefix) && ae.Level.Less(auditinternal.LevelRequest) {
```

Comment:
having knowledge about webhooks here is strange. something is wrong with the code layering.

Reply:
I'm working on associate each audit annotation value with an audit level property
```
annotations     map[string]struct{auditinternal.Level,string}
```

https://github.com/kubernetes/kubernetes/blob/9577cbfb9fd607d3698e1f2440f5875c85eca384/staging/src/k8s.io/apiserver/pkg/admission/attributes.go#L45